pkg/cloud/interfaces.go

```diff
@@ -43,9 +43,52 @@ type ClusterObject interface {
 	conditions.Setter
 }
 
+// Logger represents the ability to log messages, both errors and not.
+type Logger interface {
```

Comment:
go-logr changed the Logger interface to a struct and introduced a LogSink interface which is slightly different.

I've opted for embedding the old interface, because I think otherwise we would have to refactor a lot of code.

Not sure if that's okay / there are better alternatives

Reply:
Understood.

Reply:
Have filed #2944 to do the refactor

Reply:
Sounds good. Got a bit more lucky in the CAPI repo because we didn't embed the Logger interface in one of our interfaces. So I could just pass the struct instead of the interface around.